MGMT-12662:  Add API document for discovery kernel arguments (#4669)
openshift/assisted-service: docs/user-guide/discovery-kernel-arguments-api.md

> # Discovery Kernel Arguments API
> 
> The way to update discovery kernel arguments is by updating InfraEnv.  Currently, only **append** (additional argument) operation is supported.

Reviewer: ```suggestion
# Modifying Kernel Arguments for the Live ISO

Update the InfraEnv resource to modify the kernel arguments used in the live ISO (during the host discovery phase) .  Currently, only the **append** (additional argument) operation is supported.
```

I suggest also modifying the name of the file, people probably won't guess what "discovery kernel arguments" means.

  -H 'Content-Type: application/json' \
  -d '{
      "kernel_arguments": [
          {
            "operation": "append",
            "value": "audit=0"
          },
          {
            "operation": "append",
            "value": "trace=1"
          }
      ]
  }'
```

### During InfraEnv Update

```bash
curl -X 'PATCH' \
    'http://api.openshift.com/api/assisted-install/v2/infra-envs/3fa85f64-5717-4562-b3fc-2c963f66afa6' \
    -H 'accept: application/json' \
    -H 'Content-Type: application/json' \
    -d '{
        "kernel_arguments": [
          {
            "operation": "append",
            "value": "audit=0"
          },
          {
            "operation": "append",
            "value": "trace=1"
          }
        ]
    }'
```
When updating kernel arguments using this API call, all the kernel arguments are replaced.  So the way to clear the 
kernel arguments is just to provide empty list:
```bash
curl -X 'PATCH' \
    'http://api.openshift.com/api/assisted-install/v2/infra-envs/3fa85f64-5717-4562-b3fc-2c963f66afa6' \
    -H 'accept: application/json' \
    -H 'Content-Type: application/json' \
    -d '{
        "kernel_arguments": []
    }'
```

> ## Kube API
> 
> With Kube API, the kernel arguments are set in the InfraEnv spec section:

Reviewer: ```suggestion
## Kubernetes API

When using the Kubernetes API, the kernel arguments are set in the InfraEnv spec section:
```
